presentation/components/src/presentation-components/tree/DataProvider.ts

```diff
@@ -197,14 +207,68 @@ export class PresentationTreeDataProvider implements IPresentationTreeDataProvid
   }
 }
 
+/** @alpha */
+export interface PresentationTreeNodeItem extends DelayLoadedTreeNodeItem {
+  key: NodeKey;
+  filterInfo?: PresentationInstanceFilterInfo;
+  isFilteringDisabled?: boolean;
+  infoMessage?: string;
+}
+
+/** @alpha */
+export function isPresentationTreeNodeItem(item: TreeNodeItem): item is PresentationTreeNodeItem {
+  return (item as PresentationTreeNodeItem).key !== undefined;
+}
+
+function getFilterDefinition(imodel: IModelConnection, node?: TreeNodeItem) {
+  if (!node || !isPresentationTreeNodeItem(node) || !node.filterInfo)
+    return undefined;
+  return convertToInstanceFilterDefinition(node.filterInfo.filter, imodel);
+}
+
+function createNodesAndCountResult(nodes: Node[], count: number, parentNode?: TreeNodeItem, nodesCreateProps?: CreateTreeNodeItemProps) {
+  if (nodes.length > 0 || !parentNode || !isPresentationTreeNodeItem(parentNode) || !parentNode.filterInfo) {
+    return { nodes: createTreeNodeItems(nodes, parentNode?.id, nodesCreateProps), count };
+  }
+
+  // TODO: handle case when requesting children for node with too many children
+
+  return {
+    nodes: [createInfoNode(parentNode, translate("tree.no-filtered-children"))],
+    count: 1,
+  };
+}
+
+function createInfoNode(parentNode: PresentationTreeNodeItem, message: string): PresentationTreeNodeItem {
```

Comment:
Do we really want this info type of node to be a `PresentationTreeNodeItem`? Being a `PresentationTreeNodeItem` means it needs to have a key, which is fake doesn't make much sense. Seems like it could be an `ImmediatelyLoadedTreeNodeItem` instead.

Also, do we have a plan for "advanced" rendering of the node, e.g. a link in the middle of its text to open a dialog?

Reply:
Added separate type for `PresentationInfoTreeNodeItem`.

Regarding advanced rendering my idea is to render it in plain text when used in `PresentationTreeNodeRenderer` and have a better rendering in `PresentationTreeRenderer` as that's where we have all necessary data to open dialog, find parent node, etc.